rotors_gazebo/src/waypoint_publisher_file.cpp

```diff
@@ -120,40 +105,31 @@ int main(int argc, char** argv) {
   ros::Duration(30).sleep();
 
   ROS_INFO("Start publishing waypoints.");
-  if (!use_multi_dof_joint_trajectory) {
-    for (size_t i = 0; i < waypoints.size(); ++i) {
-      WaypointWithTime& wp = waypoints[i];
-      ROS_INFO("Publishing x=%f y=%f z=%f yaw=%f, and wait for %fs.", wp.wp.position.x, wp.wp.position.y,
-               wp.wp.position.z, wp.wp.yaw, wp.waiting_time);
-      wp.wp.header.stamp = ros::Time::now();
-      wp_pub.publish(wp.wp);
-      ros::Duration(wp.waiting_time).sleep();
-    }
-  }
-  else {
-    trajectory_msgs::MultiDOFJointTrajectoryPtr msg(new trajectory_msgs::MultiDOFJointTrajectory);
-    msg->header.stamp = ros::Time::now();
-    msg->points.resize(waypoints.size());
-    ros::Duration time_from_start(0);
-    for (size_t i = 0; i < waypoints.size(); ++i) {
-      WaypointWithTime& wp = waypoints[i];
-
-      msg->points[i].transforms.resize(1);
-      msg->points[i].velocities.resize(1);
-      msg->points[i].accelerations.resize(1);
-
-      msg->points[i].transforms[0].translation = wp.wp.position;
-      msg->points[i].transforms[0].rotation.w = cos(wp.wp.yaw * 0.5);
-      msg->points[i].transforms[0].rotation.x = 0;
-      msg->points[i].transforms[0].rotation.y = 0;
-      msg->points[i].transforms[0].rotation.z = sin(wp.wp.yaw * 0.5);
-      // We don't need velocities and accelerations here. Their constructors initialize them to zero.
-
-      msg->points[i].time_from_start = time_from_start;
-      time_from_start += ros::Duration(wp.waiting_time);
-    }
-    wp_pub.publish(msg);
+
+  trajectory_msgs::MultiDOFJointTrajectoryPtr msg(new trajectory_msgs::MultiDOFJointTrajectory);
+  msg->header.stamp = ros::Time::now();
+  msg->points.resize(waypoints.size());
+  ros::Duration time_from_start(0);
+  for (size_t i = 0; i < waypoints.size(); ++i) {
+    WaypointWithTime& wp = waypoints[i];
+
+    msg->points[i].transforms.resize(1);
+    msg->points[i].velocities.resize(1);
+    msg->points[i].accelerations.resize(1);
+
+    msg->points[i].transforms[0].translation.x = wp.position.x();
+    msg->points[i].transforms[0].translation.y = wp.position.y();
+    msg->points[i].transforms[0].translation.z = wp.position.z();
+    msg->points[i].transforms[0].rotation.w = cos(wp.yaw * 0.5);
```

Comment:
to be consistent, you could create the eigen-type with the infos you have, then use the conversion function. we didn't have it back when I implemented this.